src/main/java/roomescape/global/interceptor/LogInterceptor.java

```diff
@@ -0,0 +1,37 @@
+package roomescape.global.interceptor;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+@Component
+public class LogInterceptor implements HandlerInterceptor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LogInterceptor.class);
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+        String requestURI = request.getRequestURI();
+        LOGGER.info("[API REQUEST] {}", requestURI);
+
+        return true;
+    }
+
+    @Override
+    public void afterCompletion(HttpServletRequest request, HttpServletResponse response, Object handler, Exception ex) throws Exception {
+        String requestURI = request.getRequestURI();
+        int status = response.getStatus();
+
+        LOGGER.info("[API RESPONSE] {}: {}", requestURI, status);
+        if (status >= 200 && status < 300) {
+            LOGGER.info("[API RESPONSE] {}: {}", requestURI, status);
+        } else if (status >= 400 && status < 500) {
+            LOGGER.warn("[API RESPONSE - CLIENT ERROR] {}: {}", requestURI, status);
+        } else if (status >= 500) {
+            LOGGER.error("[API RESPONSE - SERVER ERROR] {}: {}", requestURI, status);
+        }
```

Comment:
```suggestion
        if (status >= 200 && status < 300) {
            LOGGER.info("[API RESPONSE] {}: {}", requestURI, status);
        } else if (status >= 400 && status < 500) {
            LOGGER.warn("[API RESPONSE - CLIENT ERROR] {}: {}", requestURI, status);
        } else if (status >= 500) {
            LOGGER.error("[API RESPONSE - SERVER ERROR] {}: {}", requestURI, status);
        }
```

처음으로 응답을 로깅해주는 로직은 제거해도 될 것 같습니다.
중복처럼 보이네요.

Reply:
추가로, 모든 응답을 로깅하는 것은 로그 정보가 풍부해지기도 하지만, 그만큼 단점이 될 수도 있습니다.
제프라면 어떤 방식으로 이를 세부화해볼지 고민해보면 좋을 것 같네요 🤔

Reply:
사실 아직 로깅에 대한 명확한 기준을 세우지 못한 것 같습니다. 개인적으로는 400번대나 500번대 같은 예외 상황 위주로만 로깅하는 게 가장 효율적이라고 생각합니다.

성공 상황(200번대 응답)에 대한 로깅은 대부분의 경우 별도로 확인할 필요가 없는 정상 흐름이 아닐까 생각하는데 피케이는 어떻게 생각하시나요? 추후 요청 시간 측정, API 사용 추이 분석 같은 목적이 생기면 성공 응답에 대해서도 선택적으로 로깅하거나 별도의 지표 수집 방식을 고민해 볼 수도 있을 것 같습니다.

Reply:
네, 동의합니다.
저도 필요한 로그는 요구사항으로 추가되는 것이라고 생각해서요,
추후 '이건 로그로 남겨야겠다'고 판단될 때 추가하면 된다고 봅니다.
잘 고민해주셨습니다 👍